flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/ScalarOperators.scala

```diff
@@ -861,6 +865,35 @@ object ScalarOperators {
           (l, r) => s"${qualifyMethod(BuiltInMethod.ADD_MONTHS.method)}($l, $op($r))"
         }
 
+      case (l: SqlTimeTypeInfo[_], r: SqlTimeTypeInfo[_]) if !plus =>
+        generateOperatorIfNotNull(nullCheck, LONG_TYPE_INFO, left, right) {
```

Comment:
Is always returning `LONG_TYPE_INFO` in sync with MySQL and Calcite? We need to make sure that the type information is in sync with Calcite's Sql type. Otherwise we get a type mismatch in a following operation.